presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -464,436 +477,595 @@ public void testWithJoinTables()
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a > 5", BASE_TABLE_1);
-        expectedRewrittenSql = format("SELECT a, b, c FROM %s WHERE a > 5", VIEW);
+        expectedRewrittenSql = format("SELECT a, b, c FROM %s WHERE a > 5", VIEW_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, originalViewSql);
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a = 5", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a = 4", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a = 5", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a <> 5", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a > 5", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a >= 5", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a < 3", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a = 5", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a > 5", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a > 4", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a > 3", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE c > 5", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
 
         originalViewSql = format("SELECT a, b FROM %s WHERE b = 5.0", BASE_TABLE_7);
         baseQuerySql = format("SELECT a, b FROM %s WHERE b = 5.0", BASE_TABLE_7);
-        expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 5.0", VIEW);
-
-        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
-
-        originalViewSql = format("SELECT a, b FROM %s WHERE b > 5.0", BASE_TABLE_7);
-        baseQuerySql = format("SELECT a, b FROM %s WHERE b = 5.01", BASE_TABLE_7);
-        expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 5.01", VIEW);
```

Comment:
This test was returning a false positive due to using parsing options `...setDecimalLiteralTreatment(AS_DOUBLE)` previously- in reality, it would fail to optimize on master due to a mismatch between the column types `DECIMAL(2,1)` and `DECIMAL(3,2)`. It seems reasonable that we would want to support this, so I'll create an issue shortly. For now though, removing the test.

Reply:
Use @Test(enabled = false) notation